confignode/src/main/java/org/apache/iotdb/confignode/manager/PartitionManager.java

```diff
@@ -544,8 +548,12 @@ public Map<TConsensusGroupId, Integer> getAllLeadership() {
                       regionReplicaSet.getDataNodeLocations().get(0).getDataNodeId()));
     } else {
       regionGroupCacheMap.forEach(
-          (consensusGroupId, regionGroupCache) ->
-              result.put(consensusGroupId, regionGroupCache.getLeaderDataNodeId()));
+          (consensusGroupId, regionGroupCache) -> {
+            int leaderDataNodeId = regionGroupCache.getLeaderDataNodeId();
+            if (!configManager.getNodeManager().isNodeRemoving(leaderDataNodeId)) {
+              result.put(consensusGroupId, leaderDataNodeId);
+            }
```

Comment:
We'd better put -1 instead of nothing when the leaderNode happens to be in the status of Removing. This might avoid NPE.

Reply:
fixed